contrib/mingw-cross/build.sh

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env bash
+
+#
+# exiv2/contrib/cross-ming/build.sh
+#
+# This is an "experimental" script for cross-compiling exiv2 for 64bit MinGW from Linux
+# This is derived from work discussed with Kamran Ghanaat
+# https://github.com/Exiv2/exiv2/issues/610
+#
+# Filesystem layout
+# rmills@rmillsmm-ubuntu:~/gnu/github/exiv2$ ls -l
+# drwxrwxr-x 30 rmills rmills 4096 Jan  2 12:02 exiv2
+# drwxr-xr-x 14 rmills rmills 4096 Jan  1 19:26 expat-2.2.6
+# drwxr-xr-x 16 rmills rmills 4096 Jan  1 18:48 zlib-1.2.11
+# rmills@rmillsmm-ubuntu:~/gnu/github/exiv2$
+#
+# rmills@rmillsmm-ubuntu:~/gnu/github/exiv2/exiv2/contrib/mingw-cross$ ls -l
+# -rwxrwxr-x 1 rmills rmills 2889 Jan  2 13:59 build.sh     this script
+# drwxrwxr-x 2 rmills rmills 4096 Jan  2 13:52 gcc64        DLLs
+# drwxrwxr-x 5 rmills rmills 4096 Jan  2 12:06 build        build generated in build/exiv2/bin
+# rmills@rmillsmm-ubuntu:~/gnu/github/exiv2/exiv2/contrib/mingw-cross$
+#
+# prerequisites: sudo apt install --yes realpath binutils-mingw-w64-x86-64 \
+#                g++-mingw-w64-x86-64 gcc-mingw-w64-x86-64 mingw-w64-x86-64-dev libwine-dev
+# binutils-mingw-w64-x86-64/bionic,now 2.30-7ubuntu1+8ubuntu1 amd64
+# g++-mingw-w64-x86-64/bionic,now 7.3.0-11ubuntu1+20.2build1 amd64
+# gcc-mingw-w64-x86-64/bionic,now 7.3.0-11ubuntu1+20.2build1 amd64
+# mingw-w64-x86-64-dev/bionic,bionic,now 5.0.3-1 all
+# wine64/bionic,now 3.0-1ubuntu1 amd64
+#
+# use
+# # rmills@rmillsmm-ubuntu:~/gnu/github/exiv2/exiv2/contrib/mingw-cross$ ./build.sh
+#
+
+echo PWD = $PWD
+echo this = $0
+
+COMPILER_PREFIX="x86_64-w64-mingw32"
+
+CM_GENERAL="-DCMAKE_SYSTEM_NAME=Windows                 \
+-DCMAKE_C_COMPILER=$(which ${COMPILER_PREFIX}-gcc)      \
+-DCMAKE_CXX_COMPILER=$(which ${COMPILER_PREFIX}-g++)    \
+-DCMAKE_RC_COMPILER=$(which ${COMPILER_PREFIX}-windres) \
+-DCMAKE_MAKE_PROGRAM=$(which make)               \
+-DCMAKE_FIND_ROOT_PATH="/usr/${COMPILER_PREFIX}" \
+-DCMAKE_FIND_ROOT_PATH_MODE_PROGRAM=NEVER        \
+-DCMAKE_FIND_ROOT_PATH_MODE_LIBRARY=ONLY         \
+-DCMAKE_FIND_ROOT_PATH_MODE_INCLUDE=ONLY         \
+-DCMAKE_BUILD_TYPE=Release"
+
+
+ZLIB=$( realpath ${PWD}/../../../zlib-1.2.11)
```

Comment:
There is something I do not understand about the zlb and expat dependencies. There is one point in which you mentioned that those are handled by conan, and afterwards I understood that we need to compile them by hand?

Are not these libraries handled by conan properly ?

Reply:
In fact, I see that you are installing the package `mingw64-zlib` in that fedora image.

Reply:
The Fedora build does not use this script. 

I am not sure if we even should include it at all? It builds a lot of dependencies manually, which is imho not the way to go. We can pull in the mingw dependencies via dnf or apt on Fedora, Debian and Ubuntu. Or probably via Conan on the other platforms.

Reply:
For the Linux distributions we are testing with Gitlab it makes more sense to use the system packages if possible, so that we really tests if those distributions can use Exiv2 without caring much about external dependencies.

Maybe for cross-compiling, it is more difficult to find the system packages ... and we could end up compiling something via a script or conan. 

The main aim of my comment was to highlight the contradiction in the documentation.